### PR TITLE
Add a build script for C library

### LINKED ISSUE #66: Add a build script to prepare kuliya dataset to be consumed from kuliya library
Same as we did with Rust library, a build script to generate datatypes with actual data at build time to be directly consumed by `kuliya.h`.

### PR DESCRIPTION
**Pull request by omdxp**
# Description

Resolves #66 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: c/data.h
Comment (ZibanPirate) [inline c/data.h:1]: shouldn't we gitignore this file?

File: c/helpers/string.h [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ * @param str String to be modified.
+ * @param c Characters to be removed from the given string.
+ * @returns This function do not return anything.
+ */
+void remove_chars(char *str, int c, ...)
+{
+    va_list args;
+    va_start(args, c);
+
+    while (42) // The answer for everything
```
Comment (ZibanPirate) [inline c/helpers/string.h:35]: haha why 42?

File: c/Makefile
Comment (ZibanPirate) [inline c/Makefile:1]: btw, (even on `main` branch) running `make test` will create this file: `c/test/test`.

should we put it inside a `target` folder and gitignore it?

**Review by ZibanPirate — CHANGES_REQUESTED** [review]
nice job man, took me a while to understand the code, but now i c :)

File: c/data.h
Comment (omdxp) [reply c/data.h:1]: I was thinking about this for a while and came up with these questions:

- Should the developers clone the whole repo of kuliya and `make build` to use this in their projects?
- Or should they just copy the latest changes from `c/` folder (which is `data.h` and `kuliya.h` in this case)?

So I concluded that it's easier for them for the latter option, and we the owners are responsible for updating the `data.h` header file accordingly (we can run this script whenever there is a change in `_data/` folder).

So given that, what do you think is best for the developer experience?

File: c/helpers/string.h (hunk 1 above)
Comment (omdxp) [reply c/helpers/string.h:35]: Check The Hitchhiker's Guide to the Galaxy movie and you'll get the answer 😅

File: c/Makefile
Comment (omdxp) [reply c/Makefile:1]: Ah yes! forgot about this file, I'll add it to `.gitignore`

File: c/data.h
Comment (ZibanPirate) [reply c/data.h:1]: i'm still in favour of not duplicating the code that is commited to git, i mean we're talking about C developers here, i don't think cloning a repo and running `make build` would be that inconviennt for them, also, we will be packing this code in form of libraries later so they can download the library code there instead, or, we can generate a code.zip file and attached it to the github release later.

that been said, i don't have a strong opinion here, so i'm tottaly fine with commiting it as well.

**Review by ZibanPirate — APPROVED** [review]

**Pull request merged**